ApiHelper.php

```diff
@@ -138,35 +138,32 @@ public function getMonthlyPageviews( $pages, $start, $end ) {
 					$titles[] = $r['title'];
 				}
 			}
-			// Get monthly pageviews for all of the titles i.e. original page and its redirects
+			// Get monthly pageviews for all of the titles i.e. original page and its redirects using promises
+			$promises = [];
 			foreach ( $titles as $title ) {
 				$url = 'https://wikimedia.org/api/rest_v1/metrics/pageviews/per-article/en.wikipedia/all-access/user/' . rawurlencode( $title ) . '/monthly/' . $start . '/' . $end;
-				$result = json_decode( @file_get_contents( $url ), true );
-				if ( isset( $result['items'] ) ) {
-					$results[$page] += (int)$result['items'][0]['views'];
-				} else {
-					// Report missing data
-					/*
+				$promise = $client->getAsync( $url );
+				$promises[] = $promise; // Add the promise for each request to promises array
+			}
+			try {
+				$responses = GuzzleHttp\Promise\settle( $promises )->wait();
+			} catch ( Exception $e ) {
+				// Ignore since we are logging those below anyway.
```

Comment:
The `settle` (vs. `unwrap`) should mean it will wait for all promises to finish, silently ignoring bad responses (see [docs](http://docs.guzzlephp.org/en/latest/quickstart.html#concurrent-requests)). If it for some reason `GuzzleHttp\Promise\settle( $promises )->wait()` in its entirety fails, than `$responses` will be undefined and the `foreach` below will error out, right?